Convert to v1 recipe + bump to 4.30.1

hadim opened the pull request:
<!--
Thank you for your pull request.
Below are a few things we ask you kindly to self-check before getting a review. Remove checks that are not relevant.
-->
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

<!--
Please note any issues this fixes using [closing keywords]( https://help.github.com/articles/closing-issues-using-keywords/ ):
-->

<!--
Please add any other relevant info below:
-->

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe/recipe.yaml```) and found it was in an excellent condition.

I do have some suggestions for making it better though...


For **recipe/recipe.yaml**:

 * ℹ️ PyPI default URL is now pypi.org, and not pypi.io. You may want to update the default source url.

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe/recipe.yaml```) and found it was in an excellent condition.

hadim commented:
I am seeing a linking error during the osx builds:

```
install_name_tool failed: error: changing install names or rpaths can't be redone (for architecture arm64) because larger updated load commands do not fit (the program must be relinked, and you may need to use -headerpad or -headerpad_max_install_names)
```

It happens during the linking step done by `rattler-build` and so I don't think it can really be tweaked to add `LDFLAGS="-headerpad_max_install_names ${LDFLAGS}"` for example.

hadim commented:
It's possible the issue is due to the upstream pdfium binaries that are used during the linking here: https://github.com/bblanchon/pdfium-binaries

hadim commented:
It's likely the best way to go forward here is to package `pdfium` on c-f but that looks like a complex build.

For now, I am giving up on that PR waiting for `pdfium` to be available on c-f.

See also https://github.com/conda-forge/pypdfium2-feedstock/issues/4#issuecomment-2469192619

hadim commented:
Testing again in the light of https://github.com/conda-forge/pypdfium2-feedstock/pull/8

hadim commented:
rattler-build doc for the record:

- https://rattler.build/latest/build_options/#prefix-detection-replacement-options
- https://rattler.build/latest/build_options/#dynamic-linking-configuration

hadim commented:
I confirm the build works locally on a `osx-arm64` machine with `pixi run build-osx_arm64_python3.12.____cpython --target-platform osx-arm64 --build-platform osx-arm64`

I don't know whether it's possible to enable native CI for arm64 builds on c-f (given that gha as CI is not allowed). But that seems to be the key to make it work here.

hadim commented:
From Zuilp

> Matthew R. Becker: we cannot right now due to limitations on the gha side.
> Matthew R. Becker: we'll keep you posted
> Matthew R. Becker: this is going to be a topic for the next core meeting

https://conda-forge.zulipchat.com/#narrow/channel/457337-general/topic/Native.20CI.20for.20osx-arm64.20and.20linux-aarch64/near/494223309

hadim commented:
Waiting for arm64/aarch64 GH runners on c-f, I have uploaded `linux-64`, `osx-64` and `osx-arm64` packages at https://prefix.dev/channels/pypdfium2-public

```bash
# micromamba/mamba/conda
micromamba install -c https://repo.prefix.dev/pypdfium2-public pypdfium2

# pixi
pixi project channel add https://repo.prefix.dev/pypdfium2-public
pixi add pypdfium2

# pixi exec
pixi exec --channel https://repo.prefix.dev/pypdfium2-public --channel conda-forge pypdfium2
```

hadim commented:
@jan-janssen ready for review.

- I added myself as a maintainer so I can take care of https://github.com/conda-forge/pypdfium2-feedstock/issues/9#issuecomment-2666866963 once merged here
- I converted the feedstock to v1 for faster build time
- I bumped to 4.30.1

hadim commented:
Close https://github.com/conda-forge/pypdfium2-feedstock/pull/7

hadim commented:
@conda-forge-admin, please rerender
